input-forms.md

```diff
@@ -0,0 +1,137 @@
+
+Provide access to the configured input-forms. It returns the list of existent input-forms.
+
+Example: to be provided
+
+**/api/configuration/value-pairs**   
+
+Provide access to the configured value-pairs. It returns the list of existent value-pairs.
+
+Example: to be provided
+
+## Single Input-Form 
+**/api/configuration/input-forms/<:form-name>**
+
+Provide detailed information about a specific input-form. The JSON response document is as follow
+```json
+{
+  "name": "traditional",
+  "pages": [
+  {
+  	header: {"en": "First page"},
+  	mandatory: true,
+  	fields: [
+  		{
+  			label: {"en":"Authors"},
+  			repeatable: false,
+  			mandatory: null,
+  			hints: {"en":"Enter the names of the authors of this item."},
+  			input: {
+  				type: "name"
+  			}
+  			scope: null, 
+  			visibility: {
+  				main: null,
+  				other: null
+  			},
+  			typeBind: []
+  		},
+  		{
+			label: {"en":"Title"}
+  			repeatable: false,
+  			mandatory: {"en":"You must enter a main title for this item."},
+  			hints: "Enter the main title of the item.",
+  			input: {
+  				type: "onebox",
+  				regex: null,
+  				closedVocabulary: false  				
+  			}
+  			scope: null, 
+  			visibility: {
+  				main: null,
+  				other: null
+  			},
+  			typeBind: []
+  		},
+  		...
+  	]
+  },
+  ...  
+  ],
+  "isDefault": true
+}
+
+```
+
+Exposed links:
+* value-pairs: list of value-pairs used by the specific input-form
+* collections: list of collections that explicitly use such input-form
+
+Inside the field definition we will have links if appropriate to
+* metadata: the link to the metadataregistry entry
+* authority: a definition of the authority used by the field (mainly to retrieve the autocomplete/lookup endpoint)
+* vocabulary: the link to the vocabulary xml file
+* value-pair: the list of entries for dropdown, checkbox, etc.
+ 
+## Search methods
+### findByCollection
+**/api/configuration/input-forms/search/findByCollection?uuid=<:collection-uuid>**
+
+It returns the input form that apply to a specific collection eventually fallback to the default configuration
+ 
+## Linked entities
+### Value pairs
+**/api/configuration/input-forms/<:form-name>/value-pairs**
+
+It returns the list of value-pairs used by the specific input form
+
+### collections
+**/api/configuration/input-forms/<:form-name>/collections**
+
+It returns the list of collection that make an explicit use of the input-form. If a collection doesn't specify the input-form to be used the default mapping apply but this collection is not included in the list returned by this method
+
+### metadata
+** /api/core/metadatafields/<:metadata-uuid> **
+
+It returns the metadataregistry entry associated with the field
+
+### authority
+**/api/submission/authority/<:authority-name> **
+
+It returns the authority used by the field/metadata
+
+### vocabulary
+** /api/configuration/vocabulary/<vocabylary-name> ** 
+
+It returns the vocabulary used by the field in its raw XML format
```

Comment:
I would not expose the internal implemenation details of DSpace vocabularies (XML files) to the outside world. We might want to decide to store vocabularies in the database (or authority core) if we later add the possibility for admins to add vocabulary values through the admin UI. So for now I would just convert the XML to a JSON format that can be reused if vocabularies move to the DB/SOLR.

Reply:
I agree that certain endpoints returning XML while most return JSON seems potentially problematic.  If there's a way to easily translate this to JSON that likely would be better.

Reply:
I have proposed this initial simplification because right now the vocabulary are not really managed by the backend, it is the JSP|XML UI that parse the xml. Maybe we can move this endpoint outside the api, something like /vocabulary/<:vocabulary-name>) and mark it as deprecated? If we want to rework on the vocabulary it could be better to find a way to convey the concept in the authority framework

Reply:
I have changed my mind here: we are trying to remove from the REST implementation the concepts of value-pairs and vocabularies. Both represent a special kind of authority so it should be possible to simply relate the input field to the authority that represent the value-pair or the vocabulary. Of course we will need to implement the authority plugin able to use the vocabulary XML file (an authority that read the value-pairs defined in the input-form.xml is already available in dspace).